### PR TITLE
Push built pdf to an external repo

### PR DESCRIPTION
**Pull request by qwrtln**
Marked as draft, because `publish-*` workflows should run on merge to main, not every push to the PR, but I'm testing here.

Preview of the matrix in the readme:

image

### CODE REVIEW

File: .github/workflows/publish-rule-book.yaml [hunk 1]
```diff
@@ -0,0 +1,114 @@
...
+  PO4A_VERSION: "0.71"
+  PO4A_GH_URL: https://github.com/mquinson/po4a/releases/download
+
+jobs:
+  build:
+    runs-on: ubuntu-latest
+    strategy:
+      fail-fast: false
+      matrix:
+        language: [en, pl]
```
Comment (DarthGandalf) [inline .github/workflows/publish-rule-book.yaml:21]: This line seems redundant, unless you specifically want to have elements in the matrix without Wersja

File: .github/workflows/publish-rule-book.yaml [hunk 2]
```diff
@@ -0,0 +1,114 @@
...
+
+      - uses: actions/setup-python@v5
+        if: ${{ matrix.language == 'en' }}
+        with:
+          python-version: "3.11"
+
+      - name: Add pages to select hyperlinks
+        if: ${{ matrix.language == 'en' }}
+        run: |
+          find sections -type f | xargs -n 1 sed -i "s@\\\hypertarget@\\\pagetarget@g"
```
Comment (DarthGandalf) [inline .github/workflows/publish-rule-book.yaml:82]: `find sections -type f -execdir sed -i "..." '{}' +`

Also if you replace " with ', you'll need less escaping of \

File: .github/workflows/publish-rule-book.yaml [hunk 3]
```diff
@@ -0,0 +1,114 @@
...
+
+      - name: Generate translations for ${{ matrix.language }}
+        if: ${{ matrix.language != 'en' }}
+        run: |
+          sudo apt-get install -y po4a  # required as standalone po4a is missing some deps
+          wget --quiet ${{ env.PO4A_GH_URL }}/v${{ env.PO4A_VERSION }}/po4a-${{ env.PO4A_VERSION}}.tar.gz  # sadly the version from apt isn't working
+          export PO4A_DIR="po4a-${{ env.PO4A_VERSION }}"
+          tar -xf ${PO4A_DIR}.tar.gz
+          export PERL5LIB="${PO4A_DIR}/lib:${PERL5LIB}"
+          ./${PO4A_DIR}/po4a -v --no-update po4a.cfg
```
Comment (DarthGandalf) [inline .github/workflows/publish-rule-book.yaml:49]: I see lots of code is duplicated between build* and publish* workflows.

Maybe instead add some "if" conditions on the publishing steps here, and remove the other one?

File: .github/workflows/publish-rule-book.yaml [hunk 4]
```diff
@@ -0,0 +1,114 @@
...
+          extra_system_packages: "inkscape"
+          latexmk_shell_escape: true
+          root_file: main_${{ matrix.language }}.tex
+
+      - name: Move file
+        run: |
+          mkdir ${{ matrix.language }}
+          mv main_${{ matrix.language }}.pdf ${{ matrix.language }}
+
+      # - name: Publish main_${{ matrix.language }}.pdf in artifacts repository
```
Comment (DarthGandalf) [inline .github/workflows/publish-rule-book.yaml:63]: Not needed anymore, I think

**Review by DarthGandalf — CHANGES_REQUESTED** [review]

File: .github/workflows/publish-rule-book.yaml (hunk 3 above)
Comment (qwrtln) [reply .github/workflows/publish-rule-book.yaml:49]: True, I can use different triggers for the workflow. Let me check that.

File: .github/workflows/publish-rule-book.yaml (hunk 4 above)
Comment (qwrtln) [reply .github/workflows/publish-rule-book.yaml:63]: Yeah, leftover.

File: .github/workflows/publish-rule-book.yaml (hunk 2 above)
Comment (qwrtln) [reply .github/workflows/publish-rule-book.yaml:82]: Mind making a suggestion? I'm afraid to break it xD

File: .github/workflows/publish-rule-book.yaml (hunk 1 above)
Comment (qwrtln) [reply .github/workflows/publish-rule-book.yaml]: Fixed

File: .github/workflows/publish-rule-book.yaml (hunk 3 above)
Comment (qwrtln) [reply .github/workflows/publish-rule-book.yaml]: No repetition any more.

File: .github/workflows/build-and-publish-rule-book.yaml [hunk 5]
```diff
@@ -0,0 +1,118 @@
...
+
+      - name: Add pages to select hyperlinks
+        if: ${{ matrix.language == 'en' && github.event_name != 'pull_request' }}
+        run: |
+          find sections -type f | xargs -n 1 sed -i "s@\\\hypertarget@\\\pagetarget@g"
+          python .github/insert_printable_hyperlinks.py
+
+      - name: Include index page
+        if: ${{ matrix.language == 'en' && github.event_name != 'pull_request' }}
+        run: sed -i "s@\\\include{\\\sections/back_cover.tex}@\\\include{\\\sections/index.tex}\\\include{\\\sections/back_cover.tex}@g" metadata.tex
```
Comment (DarthGandalf) [inline .github/workflows/build-and-publish-rule-book.yaml:90]:
```suggestion
        run: sed -i 's@\\include{\\sections/back_cover.tex}@\\include{\\sections/index.tex}\\include{\\sections/back_cover.tex}@g' metadata.tex
```

**Review by DarthGandalf — APPROVED** [review]

**Pull request merged**